src/index.ts

```diff
@@ -0,0 +1,196 @@
+import {
+  Component,
+  html,
+  svg,
+  render,
+  useState,
+  useEffect,
+  useRef,
+  useMemo,
+} from "uland";
+
+type AstronautData = {
+  id: number;
+  color: string;
+  power: string;
+};
+
+const randNum = () => Math.trunc(Math.random() * 10) / 10;
+const randPercent = () => `${randNum() * 100}%`;
+const randRotation = () => `${randNum() * 720}deg`;
+const randTime = () => `${1000 + randNum() * 4000}ms`;
+
+const Astronaut = Component(({ color, power }: AstronautData) => {
+  const [[x, y], updatePosition] = useState(["50%", "135px"]);
+
+  const move = () => {
+    let newX: string, newY: string;
+
+    do {
+      newX = randPercent();
+      newY = randPercent();
+    } while (x == newX && y == newY);
+
+    updatePosition([newX, newY]);
+  };
+
+  useEffect(() => {
+    setTimeout(move, 500);
+  }, []);
+
+  return html`
+    <div
+      class="user-astronaut"
+      style=${`
+        color: ${color};
+        left: ${x};
+        top: ${y};
+        transform: rotate(${randRotation()});
+        transition-duration: ${randTime()};
+      `}
+      ontransitionend=${move}
+    >
+      <svg
+        xmlns:xlink="http://www.w3.org/1999/xlink"
+        viewBox="-74 0 572 572.35609"
+      >
+        ${svg`
+        <text class="power" text-anchor="middle" x="180">
+          ${power}
+          <animate attributeName="y" dur="800ms" from="500" to="1200" repeatCount="indefinite" />
+          <animate attributeName="opacity" dur="800ms" from="1" to="0.2" repeatCount="indefinite" />
+        </text>
+        <use xlink:href="#astronaut" />
+        `}
+      </svg>
+    </div>
+  `;
+});
+
+const Astronauts = Component((ref: HTMLElement) => {
+  const fetching = useRef<boolean>(false);
+  const fetchTimeout = useRef<number>(null);
+  const updateTimeout = useRef<number>(null);
+  const [color, setColor] = useState<string>("#D0D7DF");
+  const [power, setPower] = useState<string>("💥");
+  const astronautsIndex = useRef<Record<string, null>>({});
+  const [astronauts, setAstronauts] = useState<AstronautData[]>([]);
+  const [open, setOpen] = useState<boolean>(false);
+
+  const openHatch = useMemo(
+    () => () => {
+      setOpen(true);
+      setTimeout(() => setOpen(false), 3000);
```

Comment:
FYI `setTimeout(setOpen, 3000, false)` would do the same, both `setTimeout` and `setInterval` accept extra arguments that will be passed along 👋